taichi/program/kernel.cpp

```diff
@@ -195,6 +195,9 @@ void Kernel::LaunchContextBuilder::set_arg_float(int arg_id, float64 d) {
     ctx_->set_arg(arg_id, (uint32)d);
   } else if (dt->is_primitive(PrimitiveTypeID::u64)) {
     ctx_->set_arg(arg_id, (uint64)d);
+  } else if (dt->is_primitive(PrimitiveTypeID::f16)) {
```

Comment:
Should the same modification happen in `Kernel::LaunchContextBuilder::set_arg_int`?

Reply:
@strongoier that was a bit confusing that we need to handle f32/f64 in set_arg_int... I did a quick experiment by removing them and tests passed. So I created #3384 to see if we can just remove them :P

Reply:
My understanding is that, `set_arg_int` is passing an int into the kernel, while the kernel might have a f32 argument in its signature, so we need to cast the type here.

Reply:
@strongoier in that case `set_arg_float` will be called right? https://github.com/taichi-dev/taichi/blob/master/python/taichi/lang/kernel_impl.py#L591-L599

Reply:
Ah casting also done here... Yep you're right. They can be removed.